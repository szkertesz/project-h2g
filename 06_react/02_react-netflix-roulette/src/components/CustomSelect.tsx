import Select, { components, DropdownIndicatorProps } from 'react-select';
import SelectOption from './SelectOption';
import arrowIcon from '../assets/images/icon-arrow.svg';
import { backgroundColorInput, colorAccent, colorLight, heightInput} from '../ui/CSSVariables';

interface Props {
    options: object[];
    handleChange: (e: any) => void;
}

const CustomSelect: React.FC<Props> = ({options, handleChange}) => {
    const selectOptions = options;
    const DropdownIndicator = (props: DropdownIndicatorProps) => {
        return (
            <components.DropdownIndicator {...props}>
                <img src={arrowIcon} alt='' />
            </components.DropdownIndicator>
        );
    };
    const IndicatorSeparator = () => {
        return null;
    };

    return (
        <Select
            options={selectOptions}
            isMulti
            isClearable={false}
            closeMenuOnSelect={false}
            hideSelectedOptions={false}
            onInputChange={handleChange}
            components={{
                Option: SelectOption,
                DropdownIndicator,
                IndicatorSeparator,
            }}
            styles={{
                multiValue: (base) => ({
                    ...base,
                    border: `none`,
                    backgroundColor: backgroundColorInput,
                    color: colorAccent,
                }),
                multiValueLabel: (base) => ({
                    ...base,
                    border: `none`,
                    backgroundColor: backgroundColorInput,
                    color: colorLight,
                }),
                control: (base) => ({
                    ...base,
                    height: heightInput,
                    border: `none`,
                    backgroundColor: backgroundColorInput,
                    borderRadius: 4,
                    color: 'white',
                }),
                container: (base) => ({
                    ...base,
                    height: heightInput,
                    backgroundColor: 'transparent',
                    padding: 0,
                }),
                singleValue: (base) => ({
                    ...base,
                    color: 'green',
                }),
                valueContainer: (base) => ({
                    ...base,
                    display: 'flex',
                    alignItems: 'center',
                    height: heightInput,
                    background: backgroundColorInput,
                    color: 'purple',
                    borderRadius: '4px',
                    paddingTop: '0',
                }),
                option: (base) => ({
                    ...base,
                    backgroundColor: backgroundColorInput,
                    color: colorLight,
                    display: 'flex',
                    alignItems: 'center',
                    height: '2rem',
                }),
                menu: (base) => ({
                    ...base,
                    backgroundColor: backgroundColorInput,
                }),
                input: (base) => ({
                    ...base,
                    height: heightInput,
                    margin: '0',
                    padding: '0',
                }),
            }}
        />
    );
};

export default CustomSelect