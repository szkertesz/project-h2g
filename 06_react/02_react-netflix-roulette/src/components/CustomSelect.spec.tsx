import React from 'react';
import CustomSelect from './CustomSelect';
import { render, screen } from '@testing-library/react';

describe('CustomSelect component', () => {
    // the test suite
    const options = [
        { value: 'value1', label: 'value 1 label' },
        { value: 'value2', label: 'value 2 label' },
    ];
    const handleChange = jest.fn()

    it('should render a customized Select component', () => {// the test case
        const { container } = render(<CustomSelect options={options} handleChange={handleChange} />);
        expect(container).toMatchSnapshot(); // the assertion
    });
    it('should render the CustomSelect element as it has the role \'combobox\'', () => {
        render(<CustomSelect options={options} handleChange={handleChange} />);
        expect(screen.getByRole('combobox')).toBeInTheDocument();
    })
});
