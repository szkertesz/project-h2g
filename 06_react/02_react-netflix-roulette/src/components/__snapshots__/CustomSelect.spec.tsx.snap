// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect component should render a customized Select component 1`] = `
<div>
  <div
    class=" css-1plhcwh-container"
  >
    <span
      class="css-1f43avz-a11yText-A11yText"
      id="react-select-2-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class=" css-on302j-control"
    >
      <div
        class=" css-k4907d-ValueContainer"
      >
        <div
          class=" css-14el2xx-placeholder"
          id="react-select-2-placeholder"
        >
          Select...
        </div>
        <div
          class=" css-14msk06-Input"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-describedby="react-select-2-placeholder"
            aria-expanded="false"
            aria-haspopup="true"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-2-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      >
        <div
          aria-hidden="true"
          class=" css-tlfecz-indicatorContainer"
        >
          <img
            alt=""
            src="icon-arrow.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
